src/Transport/Topology/MetaModel/NamespaceInfo.cs

```diff
@@ -2,14 +2,15 @@
 {
     public class NamespaceInfo
     {
-        public NamespaceInfo(string connectionString, NamespaceMode mode = NamespaceMode.Active)
+        public NamespaceInfo(string name, string connectionString, NamespaceMode mode = NamespaceMode.Active)
         {
+            Name = name;
             ConnectionString = connectionString;
             Mode = mode;
         }
 
+        public string Name { get; set; }
```

Comment:
Properties set via constructor perhaps should be read-only to avoid accidental changes in an existing `NamespaceInfo` objects?